lib/apiservers/engine/backends/container.go

```diff
@@ -515,6 +515,9 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 		if errModel, ok := err.(*containers.ContainerRemoveDefault); ok {
 			return derr.NewErrorWithStatusCode(fmt.Errorf("server error from portlayer : %s", errModel.Payload.Message), http.StatusInternalServerError)
 		}
+		if _, ok := err.(*containers.ContainerRemoveConflict); ok {
+			return derr.NewErrorWithStatusCode(fmt.Errorf("You cannot remove a running container. Stop the container before attempting removal or use -f"), http.StatusConflict)
+		}
 		return derr.NewErrorWithStatusCode(fmt.Errorf("server error from portlayer : %s", err), http.StatusInternalServerError)
```

Comment:
this error handling section should be moved to a `type switch` instead of a series of `if` statements see lib/apiservers/engine/backends/volume.go for an example.